Plugins/Flow.Launcher.Plugin.BrowserBookmark/Main.cs

```diff
@@ -88,35 +110,49 @@ public List<Result> Query(Query query)
 
         private static Channel<byte> refreshQueue = Channel.CreateBounded<byte>(1);
 
-        private async Task MonitorRefreshQueue()
+        private static SemaphoreSlim fileMonitorSemaphore = new(1, 1);
+
+        private static async Task MonitorRefreshQueueAsync()
         {
+            if (fileMonitorSemaphore.CurrentCount < 1)
+            {
+                return;
+            }
+            await fileMonitorSemaphore.WaitAsync();
             var reader = refreshQueue.Reader;
             while (await reader.WaitToReadAsync())
             {
-                await Task.Delay(2000);
+                await Task.Delay(5000);
                 if (reader.TryRead(out _))
                 {
-                    ReloadData();
+                    ReloadAllBookmarks();
                 }
             }
+            fileMonitorSemaphore.Release();
         }
 
         private static readonly List<FileSystemWatcher> Watchers = new();
 
         internal static void RegisterBookmarkFile(string path)
         {
             var directory = Path.GetDirectoryName(path);
-            if (!Directory.Exists(directory))
+            if (!Directory.Exists(directory) || !File.Exists(path))
+            {
                 return;
-            var watcher = new FileSystemWatcher(directory!);
-            if (File.Exists(path))
+            }
+            if (context.CurrentPluginMetadata.Disabled)
```

Comment:
this check is not necessary, `RegisterBookmarkFile` will not be called unless the plugin is enabled.

Reply:
to be clear, I mean this check:

```csharp
if (context.CurrentPluginMetadata.Disabled)
{
    return;
}
```